File: src/main/rust/aoc2022/day1.rs


fn main() {
    let input = include_str!("../../resources/aoc2022/input1.txt");
    let back: Vec<&str> = input.lines().collect();
    let mut caps: Vec<i32> = back
        .split(|s| s.is_empty())
        .map(|s| s.iter().map(|i| i.trim().parse::<i32>().unwrap()).sum())
        .collect();
    println!("caps {:?}", caps);

    let res1 = caps.iter().max().unwrap();
    println!("res1 {}", res1);

    caps.sort();
    let res2: i32 = caps.iter().rev().take(3).sum();
    println!("res2 {:?}", res2);
}

File: src/main/python/aoc2022/day1.py
def sol():
    # read file
    with open("../../resources/aoc2022/input1.txt") as f:
        elves = f.read().split("\n\n")
        calories = []
        for elf in elves:
            cals = map(int, elf.split("\n"))
            calories.append(sum(cals))
        print(f"Part 1: {max(calories)}")
        print(f"Part 2: {sum(sorted(calories, reverse=True)[:3:1])}")


if __name__ == '__main__':
    sol()
